pages/_app.js

```diff
@@ -1,10 +1,18 @@
+import { useState } from "react";
 import GlobalStyle from "@/lib/styles";
+import { activities as activityData } from "@/lib/activities";
 
 export default function App({ Component, pageProps }) {
+  const [activities, setActivities] = useState(activityData);
+
   return (
     <>
       <GlobalStyle />
-      <Component {...pageProps} />
+      <Component
+        activities={activities}
+        setActivities={setActivities}
```

Comment:
Please don't pass the complete setter function as a prop to the app. The better concept is to define the handleAddActivity here and pass this function instead.
(This is best practice because YOU want to define HOW the state can be modified e.g. if you work with other developers)